### PR TITLE
[Fix] some additional logic for re-using fasta

### PR DESCRIPTION
**Pull request by jpfeuffer**
Currently skips percolator to be able to test until ProteomicLFQ.
Also adds indexing step, if the original mzMLs were not indexed.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -68,8 +68,8 @@ if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
 ch_output_docs = Channel.fromPath("$baseDir/docs/output.md")
 
 // Validate inputs
-params.spectra = params.spectra ?: { log.error "No read data privided. Make sure you have used the '--spectra' option."; exit 1 }()
-params.database = params.database ?: { log.error "No read data privided. Make sure you have used the '--database' option."; exit 1 }()
+params.spectra = params.spectra ?: { log.error "No spectra data provided. Make sure you have used the '--spectra' option."; exit 1 }()
```
Comment (Zethson) [inline main.nf]: Oops :)
Good catch.

Comment (jpfeuffer) [reply main.nf]: hehe yes, I didnt see it in the review either.

File: nextflow.config [hunk 2]
```diff
@@ -13,6 +13,8 @@ params {
   spectra = "data/*.mzML"
   database = "data/*.fasta"
   //expdesign = "data/*.tsv"
+  adddecoys = false
+  se = "comet"
```
Comment (Zethson) [inline nextflow.config]: maybe write `se` out, since not everyone might easily understand that it's search engine

File: main.nf (hunk 1 above)
Comment (Zethson) [reply main.nf]: Funfact: This is also present in MHCquant haha

File: main.nf [hunk 3]
```diff
@@ -270,10 +337,11 @@ process percolator {
 
 }
 
+//TODO probably not needed when using Percolator. You can use the qval from there
 process fdr {
```
Comment (Leon-Bichmann) [inline main.nf]: Maybe we can here introduce:

`when:
     params.skipPercolator
`

File: main.nf [hunk 4]
```diff
@@ -296,7 +364,7 @@ process idfilter {
      file id_file from id_files_idx_feat_perc_fdr
```
Comment (Leon-Bichmann) [inline main.nf]: would then require to mix this:
`
id_files_idx_feat_perc.mix(id_files_idx_feat_perc_fdr)
`

**Pull request merged**